### PR TITLE
Fix: Issue with tagged versions and `n-v-j-l`

### LINKED ISSUE #504: `no-vulnerable-javascript-libraries` crashes the process if semver is invalid
**Site:** https://www.meetup.com/
**Connector:** chrome & jsdom

```bash 
$ npm run site -- meetup.com

> @sonarwhal/sonar@0.6.0 site /Users/alrra/projects/sonar_
> node dist/src/bin/sonar "meetup.com"

| Traversing finishedTypeError: Invalid Version: 2.0.1-pre -sizzle,-wrap,-event-alias,-effects,-dimensions,-deprecated
    at new SemVer (/Users/alrra/projects/sonar_/node_modules/semver/semver.js:293:11)
    at Range.test (/Users/alrra/projects/sonar_/node_modules/semver/semver.js:1100:15)
    at Function.satisfies (/Users/alrra/projects/sonar_/node_modules/semver/semver.js:1149:16)
    at snykInfo.reduce (/Users/alrra/projects/sonar_/dist/src/lib/rules/no-vulnerable-javascript-libraries/no-vulnerable-javascript-libraries.js:127:32)
    at Array.reduce (native)
    at detectAndReportVulnerableLibraries (/Users/alrra/projects/sonar_/dist/src/lib/rules/no-vulnerable-javascript-libraries/no-vulnerable-javascript-libraries.js:126:50)
    at <anonymous>
```

### PR DESCRIPTION
**Pull request by molant**
Fix #504

### CODE REVIEW

**Comment by molant** [comment]
@alrra can't think of a shorter description when `no-vulnerable-javascript-libraries`  is already 36 characters...

File: src/lib/rules/no-vulnerable-javascript-libraries/no-vulnerable-javascript-libraries.ts [hunk 1]
```diff
@@ -133,6 +134,20 @@ const rule: IRuleBuilder = {
             }
         };
 
+        /** Removes any tags from a version. E.g.: 2.0.0rc --> 2.0.0, 2.0.1-pre --> 2.0.1 */
+        const removeTagsFromVersion = (version: string): string => {
+            const validCharacters = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.'];
```
Comment (alrra) [inline src/lib/rules/no-vulnerable-javascript-libraries/no-vulnerable-javascript-libraries.ts]: @molant You can also use `semver for that, e.g.:

```js
const version = 'v3.4.0-rc';

console.log(`${semver.major(version)}.${semver.minor(version)}.${semver.patch(version)}`);
```

File: tests/lib/rules/no-vulnerable-javascript-libraries/tests.ts [hunk 2]
```diff
@@ -26,6 +27,10 @@ const defaultTests: Array<IRuleTest> = [
         serverConfig: generateHTMLPage(generateScriptTag(jquery))
     },
     {
+        name: `page with a library with a tagged version a no vulnerabilities (knockout 3.4.0-rc) passes`,
```
Comment (alrra) [inline tests/lib/rules/no-vulnerable-javascript-libraries/tests.ts]: **has** no vulnerabilities

File: src/lib/rules/no-vulnerable-javascript-libraries/no-vulnerable-javascript-libraries.ts (hunk 1 above)
Comment (molant) [reply src/lib/rules/no-vulnerable-javascript-libraries/no-vulnerable-javascript-libraries.ts]: Awesome 👍

File: tests/lib/rules/no-vulnerable-javascript-libraries/tests.ts (hunk 2 above)
Comment (molant) [reply tests/lib/rules/no-vulnerable-javascript-libraries/tests.ts]: Actually it is
>  and no vulnerabilties

**Comment by molant** [comment]
@alrra all changes are in

**Pull request closed without merging**